sql/sql_explain.cc

```diff
@@ -351,9 +351,9 @@ int print_explain_row(select_result_sink *result,
                       mem_root);
   
   /* 'possible_keys' */
+  StringBuffer<64> possible_keys_buf;
   if (possible_keys && !possible_keys->is_empty())
   {
-    StringBuffer<64> possible_keys_buf;
```

Comment:
This case is pretty simple to understand and fix. It's already covered by test `main.update`. And the most importantly, it's undefined behavior too.